src/core/models.py

```diff
@@ -755,16 +774,31 @@ def zookeeper_version(self) -> str:
     def broker_active(self) -> bool:
         """Checks if broker id is recognised as active by ZooKeeper."""
         broker_id = self.data_interface.local_unit.name.split("/")[1]
-        hosts = self.endpoints.split(",")
         path = f"{self.database}/brokers/ids/"
+        hosts = [host.split(":")[0] for host in self.endpoints.split(",")]
```

Comment:
suggestion: to avoid repetition, maybe this could be wrapped in a helper method?

Reply:
Totally agree, it would be fine to put in some property here, as it's still technically 'state'